src/containers/RapidAppointment/RapidAppointment.tsx

```diff
@@ -626,55 +643,39 @@ export function RapidAppointment() {
                   label="Enter Number Available (Optional)"
                   type="number"
                 />
-                <Stack distribution="fill">
-                  <Stack.Item>
-                    <TextStyle>Enter Vaccine Type (Optional)</TextStyle>
-                    <Popover
-                      active={isPopOverActive}
-                      activator={activator}
-                      onClose={() => setIsPopOverActive(!isPopOverActive)}
-                    >
-                      <ActionList
-                        items={[
-                          {
-                            content: "Pfizer",
-                            onAction: () => {
-                              setVaccineId(3);
-                              setVaccineTypeString("Pfizer");
-                              setIsPopOverActive(false);
-                            },
-                          },
-                          {
-                            content: "Moderna",
-                            onAction: () => {
-                              setVaccineId(4);
-                              setVaccineTypeString("Moderna");
-                              setIsPopOverActive(false);
-                            },
-                          },
-                          {
-                            content: "AstraZeneca",
-                            onAction: () => {
-                              setVaccineId(5);
-                              setVaccineTypeString("AstraZeneca");
-                              setIsPopOverActive(false);
-                            },
-                          },
-                          {
-                            content: "Not Sure",
-                            onAction: () => {
-                              setVaccineId(1);
-                              setVaccineTypeString("Not Sure");
-                              setIsPopOverActive(false);
-                            },
-                          },
-                        ]}
-                      />
-                    </Popover>
-                  </Stack.Item>
-                </Stack>
               </FormLayout.Group>
               <FormLayout.Group>
+                <Stack vertical>
+                  <TextStyle>Select Vaccine Types</TextStyle>
+                  <Checkbox
+                    label="Moderna"
+                    checked={isModernaChecked}
+                    onChange={() => {
+                      setIsModernaChecked(!isModernaChecked);
```

Comment:
Looks like we do this throughout this form, but if you need to mutate state based on current state, a callback should be supplied to the dispatch function to guarantee we are acting upon current state:

`setIsModernaChecked(modernaIsChecked => !modernaIsChecked);`

Reply:
Thanks!